.github/workflows/run-python-tests.yml

```diff
@@ -0,0 +1,92 @@
+        required: false
+        type: string
+      searchEngineVersion:
+        description: "Backend search engine version to use. This value must be aligned to the 'search_engine' input value."
+        required: false
+        default: 8.8
+        type: number
+      coverageReport:
+        description: "If true, a coverage report will be published."
+        required: false
+        default: false
+        type: boolean
+
+env:
+  # Increase this value to reset cache if etc/example-environment.yml has not changed
+  CACHE_NUMBER:  5
+  ALEMBIC_CONFIG: src/argilla/alembic.ini
+
+jobs:
+  run-python-tests:
+    name: Argilla python tests
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - name: Checkout Code 🛎
+        uses: actions/checkout@v2
+      - name: Setup Conda Env 🐍
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          use-mamba: true
+          activate-environment: argilla
+      - name: Get date for conda cache
+        id: get-date
+        run: echo "::set-output name=today::$(/bin/date -u '+%Y%m%d')"
+        shell: bash
+      - name: Cache Conda env
+        uses: actions/cache@v3
+        id: cache
+        with:
+          path: ${{ env.CONDA }}/envs
+          key: conda-${{ runner.os }}--${{ runner.arch }}--${{ steps.get-date.outputs.today }}-${{ hashFiles('environment_dev.yml') }}-${{ env.CACHE_NUMBER }}
+      - name: Update environment
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: mamba env update -n argilla -f environment_dev.yml
+      - name: Cache pip 👜
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ env.CACHE_NUMBER }}-${{ hashFiles('pyproject.toml') }}
+      - name: Set huggingface hub credentials
+        if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/releases')
+        env:
+          HF_HUB_ACCESS_TOKEN: ${{ secrets.HF_HUB_ACCESS_TOKEN }}
+        run: echo "Enable HF access token"
+      - name: Setup ElasticSearch 🔎
+        if: inputs.searchEngine == 'elasticsearch'
+        uses: ankane/setup-elasticsearch@v1
+        with:
+          elasticsearch-version: ${{ inputs.searchEngineVersion }}
+      - name: Setup OpenSearch 🔎
+        if: inputs.searchEngine == 'opensearch'
+        uses: ankane/setup-opensearch@v1
+        with:
+          opensearch-version: ${{ inputs.searchEngineVersion }}
+      - name: Run tests 📈
+        run: |
+          pip install -e ".[server,listeners]"
+          alembic upgrade head
+          pytest --cov=argilla --cov-report=xml ${{ inputs.pytestArgs }}
+      - name: Upload Coverage to Codecov 📦
+        uses: codecov/codecov-action@v1
+        if: inputs.coverageReport
+        with:
+          files: ./coverage.xml
+          flags: pytest
+          fail_ci_if_error: false
```

Comment:
I think that we should remove this from here, and have an step in `package.yml` that [combines](https://coverage.readthedocs.io/en/stable/cmd.html#combining-data-files-coverage-combine) the `coverage` files from both jobs (`run_tests` and `run_tests_extra`) and then upload the combined file to codecov.

Reply:
I finally could work on this with success. Now all coverage reports are combined and published automatically. So final coverage increased substantially